101-create-applicationgateway-publicip-ssl-offload/azuredeploy.json

```diff
@@ -0,0 +1,200 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "metadata": {
+                "description": "Location to deploy to"
+            }
+        },
+        "addressPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/16",
+            "metadata": {
+                "description": "Address prefix for the Virtual Network"
+            }
+        },
+        "subnetPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/28",
+            "metadata": {
+                "description": "Subnet prefix"
+            }
+        },
+        "skuName": {
+            "type": "string",
+            "allowedValues": [
+                "Standard_Small",
+                "Standard_Medium",
+                "Standard_Large"
+            ],
+            "defaultValue": "Standard_Medium",
+            "metadata": {
+                "description": "Sku Name"
+            }
+        },
+        "capacity": {
+            "type": "int",
+            "defaultValue": 2,
+            "metadata": {
+                "description": "Number of instances"
+            }
+        },
+        "backendIpAddress1": {
+            "type": "string",
+            "metadata": {
+                "description": "IP Address for Backend Server 1"
+            }
+        },
+        "backendIpAddress2": {
+            "type": "string",
+            "metadata": {
+                "description": "IP Address for Backend Server 2"
+            }
+        },
+        "certData": {
+            "type": "string",
+            "metadata": {
+                "description": "Base-64 encoded form of the .pfx file"
+            }
+        },
+        "certPassword": {
+            "type": "string",
+            "metadata": {
+                "description": "Password for .pfx certificate"
+            }
+        }
+    },
+    "variables": {
+        "applicationGatewayName": "applicationGateway1",
+        "publicIPAddressName": "publicIp1",
+        "virtualNetworkName": "virtualNetwork1",
+        "subnetName": "appGatewaySubnet",
+        "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+        "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+        "publicIPRef": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]",
+        "applicationGatewayID": "[resourceId('Microsoft.Network/applicationGateways',variables('applicationGatewayName'))]"
+    },
+    "resources": [{
+      "apiVersion": "2015-05-01-preview",
```

Comment:
Probably a good idea to make the api version a variable to move to newer versions quickly